librz/core/cmd/cmd_print.c

```diff
@@ -6599,3 +6382,219 @@ RZ_IPI RzCmdStatus rz_print_byte_bitstream_handler(RzCore *core, int argc, const
+			opts->size = core->rasm->bits / 8;
+			switch (core->rasm->bits / 8) {
+			case 1: v &= UT8_MAX; break;
+			case 2: v &= UT16_MAX; break;
+			case 4: v &= UT32_MAX; break;
+			case 8: v &= UT64_MAX; break;
+			default: break;
+			}
+			block += core->rasm->bits / 8;
+			break;
+		}
+		print_value_single(core, opts, at, v, state);
+		repeat--;
+		at += opts->size;
+	} while (repeat > 0);
+	rz_cmd_state_output_array_end(state);
+	rz_core_seek(core, old_at, false);
+	return true;
+}
+
+RZ_IPI RzCmdStatus rz_print_value_handler(RzCore *core, int argc, const char **argv, RzCmdStateOutput *state) {
+	int repeat = argc > 1 ? rz_num_math(core->num, argv[1]) : 1;
+	if (repeat < 0) {
+		return RZ_CMD_STATUS_ERROR;
+	}
+	PrintValueOptions opts = {
+		.size = 0,
+		.repeat = repeat,
+		.useBytes = false
+	};
+	print_value(core, &opts, state);
+	return RZ_CMD_STATUS_OK;
+}
+
+RZ_IPI RzCmdStatus rz_print_value1_handler(RzCore *core, int argc, const char **argv, RzCmdStateOutput *state) {
+	int repeat = argc > 1 ? rz_num_math(core->num, argv[1]) : 1;
+	if (repeat < 0) {
+		return RZ_CMD_STATUS_ERROR;
+	}
+	PrintValueOptions opts = {
+		.size = 1,
+		.repeat = repeat,
+		.useBytes = false
+	};
+	print_value(core, &opts, state);
+	return RZ_CMD_STATUS_OK;
+}
+
+RZ_IPI RzCmdStatus rz_print_value2_handler(RzCore *core, int argc, const char **argv, RzCmdStateOutput *state) {
+	int repeat = argc > 1 ? rz_num_math(core->num, argv[1]) : 1;
+	if (repeat < 0) {
+		return RZ_CMD_STATUS_ERROR;
+	}
+	PrintValueOptions opts = {
+		.size = 2,
+		.repeat = repeat,
+		.useBytes = false
+	};
+	print_value(core, &opts, state);
+	return RZ_CMD_STATUS_OK;
+}
+
+RZ_IPI RzCmdStatus rz_print_value4_handler(RzCore *core, int argc, const char **argv, RzCmdStateOutput *state) {
+	int repeat = argc > 1 ? rz_num_math(core->num, argv[1]) : 1;
+	if (repeat < 0) {
+		return RZ_CMD_STATUS_ERROR;
+	}
+	PrintValueOptions opts = {
+		.size = 4,
+		.repeat = repeat,
+		.useBytes = false
+	};
+	print_value(core, &opts, state);
+	return RZ_CMD_STATUS_OK;
+}
+
+RZ_IPI RzCmdStatus rz_print_value8_handler(RzCore *core, int argc, const char **argv, RzCmdStateOutput *state) {
+	int repeat = argc > 1 ? rz_num_math(core->num, argv[1]) : 1;
+	if (repeat < 0) {
+		return RZ_CMD_STATUS_ERROR;
+	}
+	PrintValueOptions opts = {
+		.size = 8,
+		.repeat = repeat,
+		.useBytes = false
+	};
+	print_value(core, &opts, state);
+	return RZ_CMD_STATUS_OK;
+}
+
+RZ_IPI RzCmdStatus rz_print_value_asciiz_handler(RzCore *core, int argc, const char **argv, RzCmdStateOutput *state) {
+	if (argc < 2) {
+		// TODO: Use the API
+		rz_core_cmdf(core, "ps");
+		return RZ_CMD_STATUS_OK;
+	}
+	const char *stack[] = {
+		"ret", "arg0", "arg1", "arg2", "arg3", "arg4", NULL
+	};
+	int n = core->rasm->bits / 8;
+	for (int i = 0; stack[i]; i++) {
+		if (!strcmp(argv[1], stack[i])) {
+			// TODO: Use the API
+			rz_core_cmdf(core, "ps @ [`drn sp`+%d]", n * i);
+		}
+	}
+	return RZ_CMD_STATUS_OK;
+}
```

Comment:
can we kill this functionality? it is not tested, it is not explained in the doc and it is unclear :(

Reply:
With pleasure.